From 5d19080c5104f42d39cd3a0f5010707d53b3a82a Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Tue, 12 Dec 2017 13:08:36 +0200
Subject: [PATCH 3124/5242] MLK-17231-3: drm: imx: dcss: drop runtime
 autosuspend feature

commit  ed5efac67e5642bd04ce98a375f80833c05148e3 from
https://source.codeaurora.org/external/imx/linux-imx.git

This was needed when using the mode_set_nofb() callback. Since
everything was moved to crtc_enable() callback, runtime autosuspend
can be dropped.

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/dcss/dcss-crtc.c |    3 +--
 drivers/gpu/imx/dcss/dcss-common.c   |    2 --
 2 files changed, 1 insertion(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/imx/dcss/dcss-crtc.c b/drivers/gpu/drm/imx/dcss/dcss-crtc.c
index 9931aba..fd758f7 100644
--- a/drivers/gpu/drm/imx/dcss/dcss-crtc.c
+++ b/drivers/gpu/drm/imx/dcss/dcss-crtc.c
@@ -202,8 +202,7 @@ static void dcss_crtc_atomic_disable(struct drm_crtc *crtc,
 	wait_for_completion_timeout(&dcss_crtc->disable_completion,
 				    msecs_to_jiffies(100));
 
-	pm_runtime_mark_last_busy(dcss_crtc->dev->parent);
-	pm_runtime_put_autosuspend(dcss_crtc->dev->parent);
+	pm_runtime_put_sync(dcss_crtc->dev->parent);
 }
 
 static const struct drm_crtc_helper_funcs dcss_helper_funcs = {
diff --git a/drivers/gpu/imx/dcss/dcss-common.c b/drivers/gpu/imx/dcss/dcss-common.c
index cd4afdc..178383f 100644
--- a/drivers/gpu/imx/dcss/dcss-common.c
+++ b/drivers/gpu/imx/dcss/dcss-common.c
@@ -438,8 +438,6 @@ static int dcss_probe(struct platform_device *pdev)
 
 	dcss_debugfs_init(dcss);
 
-	pm_runtime_set_autosuspend_delay(&pdev->dev, 3000);
-	pm_runtime_use_autosuspend(&pdev->dev);
 	pm_runtime_enable(&pdev->dev);
 
 	dcss_bus_freq(dcss, true);
-- 
1.7.9.5

